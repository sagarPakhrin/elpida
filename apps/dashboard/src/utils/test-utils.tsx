import { render, RenderOptions } from '@testing-library/react';
import React, { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { UiProvider } from '../context/ui-context';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <MemoryRouter>
      <UiProvider>{children}</UiProvider>
    </MemoryRouter>
  );
};

const customRender = (ui: ReactElement, options: RenderOptions = {}) =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render, render as rtlRender };
