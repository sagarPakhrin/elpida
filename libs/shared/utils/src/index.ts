export * from './lib/format-currency';
export * from './lib/int-to-string';
