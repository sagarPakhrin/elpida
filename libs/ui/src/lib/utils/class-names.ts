// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const classNames = (...classes: any) => {
  return classes.filter(Boolean).join(' ');
};
