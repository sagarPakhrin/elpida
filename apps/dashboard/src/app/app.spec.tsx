import { render } from '../utils/test-utils';

import App from './app';

describe('App', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<App />);

    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });

  it('should have a greeting as the title', () => {
    const { getByText } = render(<App />);

    expect(getByText(/Welcome to dashboard/i)).toBeTruthy();
  });
});
