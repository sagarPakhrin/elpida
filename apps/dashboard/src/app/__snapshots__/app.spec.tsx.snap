// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render successfully 1`] = `
<body>
  <div>
    <div
      class="h-screen w-full flex bg-gray-50"
    >
      <aside
        class="py-3 h-full border-r bg-teal-900  text-gray-300  border-gray-200 flex flex-col items-center justify-between transition-all duration-300 w-64"
      >
        <div
          class="w-full"
        >
          <a
            class="flex mx-4 px-4 justify-center text-2xl pb-2 font-semibold text-gray-500 text-center border-b border-gray-200"
            href="/"
          >
            <span
              class=""
            >
              Constructor
            </span>
          </a>
          <div
            class="pt-6 px-1 space-y-1"
          >
            <a
              aria-current="page"
              class="flex py-3 px-4 rounded hover:bg-teal-800 gap-3 text-sm items-center bg-teal-800"
              href="/dashboard"
            >
              <svg
                aria-hidden="true"
                class="flex-shrink-0 h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 3v11.25A2.25 2.25 0 006 16.5h2.25M3.75 3h-1.5m1.5 0h16.5m0 0h1.5m-1.5 0v11.25A2.25 2.25 0 0118 16.5h-2.25m-7.5 0h7.5m-7.5 0l-1 3m8.5-3l1 3m0 0l.5 1.5m-.5-1.5h-9.5m0 0l-.5 1.5M9 11.25v1.5M12 9v3.75m3-6v6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              Dashboard
            </a>
            <a
              class="flex py-3 px-4 rounded hover:bg-teal-800 gap-3 text-sm items-center"
              href="/products"
            >
              <svg
                aria-hidden="true"
                class="flex-shrink-0 h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.75 10.5V6a3.75 3.75 0 10-7.5 0v4.5m11.356-1.993l1.263 12c.07.665-.45 1.243-1.119 1.243H4.25a1.125 1.125 0 01-1.12-1.243l1.264-12A1.125 1.125 0 015.513 7.5h12.974c.576 0 1.059.435 1.119 1.007zM8.625 10.5a.375.375 0 11-.75 0 .375.375 0 01.75 0zm7.5 0a.375.375 0 11-.75 0 .375.375 0 01.75 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              Products
            </a>
          </div>
        </div>
        <div
          class="w-full"
        >
          <button
            class="w-full flex py-3 px-4 rounded gap-3 text-sm items-center hover:bg-teal-800"
          >
            <svg
              aria-hidden="true"
              class="flex-shrink=0 h-6 w-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.75 9V5.25A2.25 2.25 0 0013.5 3h-6a2.25 2.25 0 00-2.25 2.25v13.5A2.25 2.25 0 007.5 21h6a2.25 2.25 0 002.25-2.25V15m3 0l3-3m0 0l-3-3m3 3H9"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            Logout
          </button>
        </div>
      </aside>
      <div
        class="flex flex-1 flex-col overflow-y-auto"
      >
        <header
          class="flex justify-between py-3 px-10 shadow text-gray-500 bg-white sticky z-20"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              type="button"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </div>
          <div
            class="flex gap-4 items-center"
          >
            <button
              type="button"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.25 9a6.75 6.75 0 0113.5 0v.75c0 2.123.8 4.057 2.118 5.52a.75.75 0 01-.297 1.206c-1.544.57-3.16.99-4.831 1.243a3.75 3.75 0 11-7.48 0 24.585 24.585 0 01-4.831-1.244.75.75 0 01-.298-1.205A8.217 8.217 0 005.25 9.75V9zm4.502 8.9a2.25 2.25 0 104.496 0 25.057 25.057 0 01-4.496 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <button
              type="button"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M18.685 19.097A9.723 9.723 0 0021.75 12c0-5.385-4.365-9.75-9.75-9.75S2.25 6.615 2.25 12a9.723 9.723 0 003.065 7.097A9.716 9.716 0 0012 21.75a9.716 9.716 0 006.685-2.653zm-12.54-1.285A7.486 7.486 0 0112 15a7.486 7.486 0 015.855 2.812A8.224 8.224 0 0112 20.25a8.224 8.224 0 01-5.855-2.438zM15.75 9a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </header>
        <div
          class="flex-1"
        >
          <div>
            <h1>
              Welcome to Dashboard
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
